uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/shared/components/leave-calendar.component.js

```diff
@@ -266,59 +282,123 @@ define([
+        .then(source === 'period' ? setFirstPeriodMonth : _.noop)
         .then(source === 'contacts' ? loadContacts : _.noop)
+        .then(source === 'month' ? setMonthPaginatorsAvailability : _.noop)
         .then(function () {
-          injectAndShowMonths((source === 'contacts'));
+          injectAndShowMonth((source === 'contacts'));
         });
     }
 
+    function selectCurrentMonth () {
+      var selectedPeriodId = vm.selectedPeriod.id;
+
+      vm.selectedPeriod = _.find(vm.absencePeriods, function (period) {
+        return !!period.current;
+      });
+
+      (selectedPeriodId !== vm.selectedPeriod.id) && buildPeriodMonthsList();
+      setCurrentMonth();
+      setMonthPaginatorsAvailability();
+      refresh('month');
+    }
+
     /**
      * Sends the "show" signal to the leave-calendar-month components
      *
      * @param {Boolean} forceDataReload if true, then a month will load its data
      *   regardless if it had already loaded it
      */
-    function sendShowMonthsSignal (forceDataReload) {
-      var monthsToShow = !vm.selectedMonths.length
-        ? vm.months
-        : vm.months.filter(function (month) {
-          return _.includes(vm.selectedMonths, month.index);
-        });
+    function sendShowMonthSignal (forceDataReload) {
+      $rootScope.$emit('LeaveCalendar::showMonth', !!forceDataReload);
+    }
+
+    /**
+     * Sets the month that is to be selected by default
+     */
+    function setCurrentMonth () {
+      vm.selectedMonthIndex = moment().year() + '-' + moment().month();
+
+      setSelectedMonth();
+    }
+
+    /**
+     * Sets the first month from the currently selected period as the selected month
+     */
+    function setFirstPeriodMonth () {
+      vm.selectedMonthIndex = vm.months[0].index;
 
-      $rootScope.$emit('LeaveCalendar::showMonths', monthsToShow, !!forceDataReload);
+      setSelectedMonth();
+    }
+
+    /**
+     * Enables or disables the month paginator of a specified direction.
+     * It disables the paginator if there are no months to paginate to.
+     *
+     * @param {String} direction previous|next
+     */
+    function setMonthPaginatorAvailability (direction) {
+      var edgeMonthSelector = direction === 'previous' ? 'first' : 'last';
+      var edgeMonth = _[edgeMonthSelector](vm.months);
+      var edgeMonthMoment = moment().year(edgeMonth.year).month(edgeMonth.month);
+
+      vm.monthPaginatorsAvailability[direction] =
+        !vm.selectedMonth.moment.isSame(edgeMonthMoment, 'month');
+    }
+
+    /**
+     * Enables or disables the month paginators
+     */
+    function setMonthPaginatorsAvailability () {
+      setMonthPaginatorAvailability('previous');
+      setMonthPaginatorAvailability('next');
     }
 
     /**
-     * Sets the months that are to be selected by default
+     * Sets the month that is was selected
      */
-    function setDefaultMonths () {
-      vm.selectedMonths = [moment().month()];
+    function setSelectedMonth () {
```

Comment:
I find `setSelectedMonth` a bit confusing because you don't pass any parameters, it just uses `vm.selectedMonthIndex` so it's always a two step process:

```js
vm.selectedMonthIndex = getMonthIndex(someMonth);
setSelectedMonth();
```

Can we do this instead:

```js
function setSelectedMonthAndIndex(momentMonth) {
  vm.selectedMonthIndex = getMonthIndex(someMonth);
  vm.selectedMonth = _.find(vm.months, { index: vm.selectedMonthIndex });
}
```

Suggestions are valid.

Reply:
Done.